src/protocols/MeterS0.cpp

```diff
@@ -238,31 +238,34 @@ void MeterS0::counter_thread()
 		if (is_blocking) {
 			bool timeout = false;
 			if (_hwif->waitForImpulse( timeout )) {
+				//Event erkannt. Wert des Events ist, aufgrund des prellen reiner Zufall!!
+				//deswegen erst entprellen .....
```

Comment:
Woran wird das zappeln der Leitung erkannt?
Ich denke, dass der GPIO Pin kontinuierlich gepollt wird. Sobald der Wert sich vom letzten Wert unterscheidet, muss sich was am Pin getan haben. Jetzt kann man annehmen, dass das zappeln beginnt.

Die Pollschleife ist der Counter Tread. Dieser testet kontinuierlich "waitForImpulse". In waitForImpulse ist der eigentliche poll für maximal 1s auf die Schnittstelle. rv>0 sagt, es ist was am Eingang passiert.  if (poll_fd.revents & POLLPRI) muss erfüllt sein, sonst würden keine Impulse gezählt werden können. (Ich kenne die poll Funktion nicht und weis nicht, was revents, bzw POLLPRI ausdrücken soll) Auch dieses IF `if (::pread(_fd, buf, 1, 0) < 1` verstehe ich nicht. Ich kann aber sicher sagen, das rv>0 ist, wenn eine Flanke von 0 nach 1 oder 1 nach 0 stattfindet. Dann wird ein "TRUE" zurückgeliefert. Die Auswertung, dass es sich tatsächlich ausschlieslich um eine steigenden Flanke handelt kann ich nicht erkennen (dies wäre aber für einen positiven Impuls erforderlich) Auch liest die poll Funktion eventuell nicht nur einen Wert in den poll_fd, sondern eventuell aufgrund des prellens auch deutlich mehr, wie erwartet. 
Fakt bei meiner Eingangsbeschaltung war, dass konsequent fast 90% aller Flankenwechsel (positive, wie negative) als Impulse gezählt wurden. Jetzt werden durch die Prüfung auf den Wert auf >0 mittels _hwif->status() nur dann die Impulse gezählt, wenn ein waitForImpulse erkannt wurde und nach der debounce Time der Wert am GPIO auch logisch 1 ist. Seit dieser Änderung zählt mein vzlogger nur noch positive Flanken (sprich pro Umdrehung nur noch 1 statt wie bisher 2)

Es gab bislang nicht immer (aber sporadisch) die doppelte Frequenz bei langsammen Pegelwechsel. Das Problem liegt am prellen des Eingangs. Der Wert des GPIO nimmt von sicher 0 nach sicher 1 während der Prellzeit nahezu unendlich viele Zwischenwerte an. Während der Prellzeit können theoretisch also bereits "unendlich" (praktisch ist diese Zahl auf die Abtastrate des GPIO begrenzt) viele Flanken auftreten. Welche von den vielen Flanken der poll erwischt ist reiner Zufall. Als waitforImpulse dürften Flanken aber eigentlich nur gewertet werden, wenn der GPIO vor dem zappeln sicher 0 und nach dem zappeln sicher 1 ist. Diese Auswertung kann ich nicht erkennen. Deswegen kommt es recht häufig (bei meiner Schaltung nahezu immer) auch zu einem Impulse bei eigentlich negativer Flanke.

Wie du richtig in diesem Satz erkannt hast:

> D.h. wenn innerhalb Debounce Zeit der Wert von 0 auf 1 und wieder auf 0 fällt, dann wird nicht gezählt.

ist genau dieses zählen während der Prellzeit zu verhindern. 

Hieraus folgt auch, dass der minimale Impulsabstand des Zählers > der Debounce Zeit sein muss, um überhaupt sicher zählen zu können.